packages/flutter_tools/lib/src/windows/migrations/version_migration.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../cmake_project.dart';
+
+const String _cmakeFileBefore =
+'# Apply the standard set of build settings. This can be removed for applications\r\n'
```

Comment:
So we're sure this will end in `\r\n` because when you use git on Windows, and it checks out https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/app_shared/windows.tmpl/CMakeLists.txt.tmpl, it will auto convert `\n` -> `\r\n`, or is there some other OS-level magic? Or are there some configurations of git that might lead to this file having only `\n` between lines?

Reply:
Or, do we know this must have `\r\n` because otherwise windows cmake will explode?

Reply:
Good question. Somehow this has `\r\n` line endings on my computer, let me investigate how that happens.

Reply:
It turns out it's because I've enabled `autocrlf` globally on git. This converts my Flutter SDK's template files from `LF` to `CRLF` on checkout. It's likely most users have `LF` line endings, so we should support that as well.

Reply:
I'm gonna be out for a week, however this change LGTM provided we're confident it will work no matter which settings they have in their windows git config :)